### PR TITLE
Fix user import when it overlaps with runtime library modules

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/1737.

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
There could be other approaches to fix https://github.com/lcompilers/lpython/issues/1737, for example, changing/reversing the priority of `runtime library` path and `parent_dir` or using name mangling. 

This PR fixes the issue by pruning the paths to be searched (as we need not re-search the package/module).

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please review and share feedback.

**Comment by Thirumalai-Shaktivel** [comment]
LGTM!

**Review by certik — APPROVED** [review]
I think it's fine. @czgdp1807 can you also please review?

**Review by czgdp1807 — APPROVED** [review]

File: integration_tests/test_import/sys.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+def hi_from_user_sys():
+    print("hi from user sys!")
```
Comment (czgdp1807) [inline integration_tests/test_import/sys.py:2]: Probably return an integer value from here.

File: integration_tests/test_import_05.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from test_import.sys import hi_from_user_sys
+
+hi_from_user_sys()
```
Comment (czgdp1807) [inline integration_tests/test_import_05.py:3]: And then assert here. So that we know that we are not having any hidden bugs.

Comment (Shaikh-Ubaid) [reply integration_tests/test_import_05.py]: Done! Thanks for the guidance!

**Comment by Shaikh-Ubaid** [comment]
I am adding this to auto-merge as it seems approved.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**